scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -590,8 +590,9 @@ class Router(formatOps: FormatOps) {
           .flatMap(templateCurly)
           .getOrElse(leftOwner.tokens.last)
         val forceNewlineBeforeExtends = Policy.before(expire) {
-          case Decision(t @ FormatToken(_, _: T.KwExtends, _), s)
-              if t.meta.rightOwner == leftOwner =>
+          case Decision(t @ FormatToken(_, right, _), s)
+              if (right.is[T.KwExtends] || right.is[soft.KwDerives]) &&
```

Comment:
optional: should this check be added as a trait to `SoftKeywordClasses` within `util/TokenClasses.scala`? it's used several times.

Reply:
Added a new trait there, good idea.